src/Build/Logging/BaseConsoleLogger.cs

```diff
@@ -642,39 +644,38 @@ internal SortedList ExtractItemList(IEnumerable items)
         /// </summary>
         internal virtual void OutputItems(string itemType, ArrayList itemTypeList)
         {
-            // Write each item, one per line
-            bool haveWrittenItemType = false;
+            setColor(ConsoleColor.Gray);
+            WriteLinePretty(itemType);
             setColor(ConsoleColor.DarkGray);
-            foreach (ITaskItem item in itemTypeList)
```

Comment:
Earlier we only used to get items of type `ITaskItem` (from `ProjectStartedEventArgs`). Now we can also get instances of `ProjectItem` (from `ProjectEvaluationFinishedEventArgs`), which don't implement either `ITaskItem` or `IMetadataContainer`. Hence some ugly pattern-matching and special-casing. Made me wonder whether `ProjectItem` should implement `ITaskItem` and/or `IMetadataContainer`.

Also this behavior seems to break the test `TestItemsWithUnexpandableMetadata` (https://github.com/dotnet/msbuild/issues/6518)